vault/data_source_transit_secret_backend_key.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func transitSecretBackendKeyDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: provider.ReadContextWrapper(readTransitSecretBackendKey),
+		Schema: map[string]*schema.Schema{
+			consts.FieldBackend: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Full path where transit backend is mounted.",
+			},
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the key.",
+			},
+			"type": {
```

Comment:
There are a few keys missing from this

- auto_rotate_period
- imported_key (it's listed as imported below)
- latest_version
- min_available_version

There are also a few optional keys depending on options set in the response returned see [formatKeyPolicy](https://github.com/hashicorp/vault/blob/main/builtin/logical/transit/path_keys.go#L335)

- key_size is returned if set on creation
- imported_key_allow_rotation is returned if imported_key is true
- backup_info and restore_info are returned if the key was backed up/restored
- kdf, kdf_mode, convergent_encryption and possibly convergent_encryption_version are returned if derived is true

Reply:
> There are a few keys missing from this

Sorry I used an old vault release without some of these keys and after testing it with the new version I forgot to add them.